runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/graph/CreateExecutableStageNodeFunction.java

```diff
@@ -191,6 +195,37 @@ public Node apply(MutableNetwork<Node, Edge> input) {
         Iterables.filter(input.nodes(), InstructionOutputNode.class)) {
       InstructionOutput instructionOutput = node.getInstructionOutput();
 
+      // If this is the input PCollection or the output PCollection for an ExecutableStage, it's
+      // necessary to check whether the window coder is not a GlobalWindow coder.
+      if (isExecutableStageInputPCollection(input, node)
+          || isExecutableStageOutputPCollection(input, node)) {
+        Coder<?> javaCoder =
+            CloudObjects.coderFromCloudObject(CloudObject.fromSpec(instructionOutput.getCodec()));
+        // For now, Dataflow runner harness only deal with FixedWindow.
+        if (FullWindowedValueCoder.class.isInstance(javaCoder)) {
+          FullWindowedValueCoder<?> windowedValueCoder = (FullWindowedValueCoder<?>) javaCoder;
+          Coder<?> windowCoder = windowedValueCoder.getWindowCoder();
+          if (IntervalWindowCoder.class.isInstance(windowCoder)) {
+            fakeWindowingStrategyId = "generatedFixedWindowingStrategy" + idGenerator.getId();
+            try {
+              // Since the coder is the only needed from a WindowingStrategy, the size field of one
+              // FixedWindows is meaningless here.
+              RunnerApi.MessageWithComponents windowingStrategyProto =
+                  WindowingStrategyTranslation.toMessageProto(
+                      WindowingStrategy.of(FixedWindows.of(Duration.standardSeconds(1))),
+                      sdkComponents);
+              componentsBuilder.putWindowingStrategies(
+                  fakeWindowingStrategyId, windowingStrategyProto.getWindowingStrategy());
+              componentsBuilder.putAllCoders(windowingStrategyProto.getComponents().getCodersMap());
+              componentsBuilder.putAllEnvironments(
```

Comment:
Do we need this? Wouldn't this break for non-Java?

Reply:
I didn't test these against non-java but I guess there is a huge possibility to break non-java. This is another concern, I may not be able to extract window coder and element coder from a non-java coder.